crates/scope/src/context.rs

```diff
@@ -610,17 +613,762 @@ impl BlockContext {
     }
 }
 
+/// The name of function expression.
+#[derive(Debug)]
+struct FunctionExpressionContext {
+    function_expression_name: Option<SourceAtomSetIndex>,
+
+    scope_index: ScopeIndex,
+
+    name_tracker: FreeNameTracker,
+}
+
+impl FunctionExpressionContext {
+    fn new(scope_index: ScopeIndex) -> Self {
+        Self {
+            function_expression_name: None,
+            scope_index,
+            name_tracker: FreeNameTracker::new(),
+        }
+    }
+
+    fn set_function_name(&mut self, name: SourceAtomSetIndex) {
+        self.function_expression_name = Some(name);
+    }
+
+    fn into_scope_data(self, enclosing: ScopeIndex) -> ScopeData {
+        match &self.function_expression_name {
+            Some(name) => {
+                // Runtime Semantics: Evaluation
+                // https://tc39.es/ecma262/#sec-function-definitions-runtime-semantics-evaluation
+                //
+                // FunctionExpression :
+                //   function BindingIdentifier ( FormalParameters )
+                //   { FunctionBody }
+                //
+                // Step 1. Let scope be the running execution context's
+                //         LexicalEnvironment.
+                // Step 2. Let funcEnv be NewDeclarativeEnvironment(scope).
+                // Step 3. Let envRec be funcEnv's EnvironmentRecord.
+                let mut data = LexicalScopeData::new_named_lambda(enclosing);
+
+                // Step 4. Let name be StringValue of BindingIdentifier .
+                // Step 5. Perform envRec.CreateImmutableBinding(name, false).
+                let is_closed_over = false;
```

Comment:
looks like I forgot to implement this.
this should check name_tracker, like others.

Reply:
Easy fix.